README.md

```diff
@@ -69,30 +69,25 @@ Create an executable file in the .git/hooks directory of the target repository:
 
 ```bash
 #!/bin/bash
-if [ -d ".github/workflows" ]; then
-  if command -v action-validator >/dev/null 2>&1; then
-    echo "Running pre-commit hook for GitHub Actions: https://github.com/mpalmer/action-validator"
-    scan_count=0
-    for action in $(find .github/workflows -name "*.y*ml"); do
-      validate="$(action-validator "$action")"
-      if [ -z "$validate" ]; then
-        echo "✅ $action"
-      else
-        echo "❌ $action"
-        echo "$validate"
-        exit 1
-      fi
-      scan_count=$((scan_count+1))
-    done
-    echo "action-validator scanned $scan_count GitHub Actions found no errors!"
+if ! command -v action-validator >/dev/null; then
+  echo "action-validator is not installed."
+  echo "Installation instructions: https://github.com/mpalmer/action-validator"
+  exit 1
+fi
+echo "Running pre-commit hook for GitHub Actions: https://github.com/mpalmer/action-validator"
+scan_count=0
+for action in $(git diff --cached --name-only --diff-filter=ACM | grep -E "\.github/workflows|actions" | grep -E "\.ya*ml$"); do
+  validate="$(action-validator "$action")"
+  if $validate; then
```

Comment:
Does that work?  I don't think I've ever actually tried to just smash a command output into an `if` statement.

This should, in theory, work at least as well:

```
if [ "$?" = "0" ]; then
```

Reply:
This syntax does work; I've tried some valid and invalid action yamls. I tried your syntax first but got a shellcheck error:
https://github-wiki-see.page/m/koalaman/shellcheck/wiki/SC2181

Reply:
`if $validate` requires the output of `action-validator "$action"` to be a valid Bash command, which I suspect is highly unlikely. You probably want `if [[ "$?" -eq 0 ]]; then`.